File: server/src/models/db/map-stats.js
'use strict';

module.exports = (sequelize, type) => {
	return sequelize.define('mapStats', {
		totalReviews: {
			type: type.INTEGER.UNSIGNED,
			defaultValue: 0,
		},
		totalDownloads: {
			type: type.INTEGER.UNSIGNED,
			defaultValue: 0,
		},
		totalSubscriptions: {
			type: type.INTEGER.UNSIGNED,
			defaultValue: 0,
		},
		totalPlays: {
			type: type.INTEGER.UNSIGNED,
			defaultValue: 0,
		},
		totalFavorites: {
			type: type.INTEGER.UNSIGNED,
			defaultValue: 0,
		},
		totalCompletions: {
			type: type.INTEGER.UNSIGNED,
			defaultValue: 0,
		},
		totalUniqueCompletions: {
			type: type.INTEGER.UNSIGNED,
			defaultValue: 0,
		},
		totalTimePlayed: {
			type: type.BIGINT.UNSIGNED,
			defaultValue: 0,
		},
	})
};

File: API/src/Modules/Maps/Momentum.Maps.Core/Models/MapStats.cs
namespace Momentum.Maps.Core.Models
{
    public class MapStats
    {
        public BaseStats BaseStats { get; set; }
        public uint TotalReview { get; set; }
        public uint TotalDownloads { get; set; }
        public uint TotalSubscriptions { get; set; }
        public uint TotalPlays { get; set; }
        public uint TotalFavorites { get; set; }
        public uint TotalCompletions { get; set; }
        public uint TotalUniqueCompletions { get; set; }
        // TODO: Currently unused in the staging server
        public uint TotalTimePlayed { get; set; }
    }
}